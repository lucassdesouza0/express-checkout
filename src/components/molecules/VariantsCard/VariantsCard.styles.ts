import styled from "@emotion/styled";

export const Card = styled("div")`
  flex: 1;
  flex-direction: column;
  display: flex;
`;
