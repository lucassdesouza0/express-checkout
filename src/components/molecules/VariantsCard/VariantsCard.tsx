import React, { useContext, useState } from "react";
import { useTranslation } from "next-i18next";
import { useRouter } from "next/router";
import { ContextProps, ProductProps } from "pages/_app";

import Selection from "atoms/Selection/Selection";
import Button from "atoms/Button/Button";

import { Card } from "./VariantsCard.styles";

export interface VariantsCardProps {
  product?: ProductProps;
}

const VariantsCard = ({ product }: VariantsCardProps) => {
  const { t } = useTranslation("common");
  const router = useRouter();

  return (
    <Card>
      <Selection product={product} />
      <Button
        type="button"
        onClick={() => router.push("checkout")}
        disabled={false}
      >
        <a href="/checkout">{t("kart")}</a>
      </Button>
    </Card>
  );
};

export default VariantsCard;
