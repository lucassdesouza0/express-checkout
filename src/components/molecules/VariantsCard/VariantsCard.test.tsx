import React from "react";
import { screen, render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import testRender from "../../../tests/render";

import mock from "../../../tests/__mocks__/product";

import VariantsCard from "./VariantsCard";

describe("<ProductCard />", () => {
  test("it should mount", () => {
    testRender(<VariantsCard product={mock}></VariantsCard>);

    const card = screen.getAllByTestId("variants-card");

    expect(card[0]).toBeInTheDocument();
  });

  test("it should render the load version of the card", () => {
    const { asFragment } = render(<VariantsCard />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("it should match snapshot", () => {
    const { asFragment } = render(<VariantsCard product={mock}></VariantsCard>);
    expect(asFragment()).toMatchSnapshot();
  });
});
