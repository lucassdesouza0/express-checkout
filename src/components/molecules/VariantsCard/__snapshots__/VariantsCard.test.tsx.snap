// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard /> it should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1ooee9y-Card e1a1q5s70"
    data-testid="variants-card"
  >
    <div
      class="css-1px0wn7-Section eui42cl2"
    >
      <div
        class="css-1l33m80-SelectionSection eui42cl1"
      >
        <label>
          packing-size:
        </label>
        <select
          class="css-1nbhqyd-Select eui42cl0"
          name="ages"
        >
          <option
            data-testid="select-option"
            disabled=""
            selected=""
            value="sizes"
          >
            sizes
          </option>
          <option
            data-testid="select-option-3kg"
            label="3kg"
            value="0"
          >
            3kg
          </option>
          <option
            data-testid="select-option-10kg"
            label="10kg"
            value="1"
          >
            10kg
          </option>
          <option
            data-testid="select-option-20kg"
            label="20kg"
            value="2"
          >
            20kg
          </option>
        </select>
      </div>
    </div>
    <button
      class="css-hhtkyx-Button enhitpr0"
      name="button"
      type="button"
    >
      <a
        href="/checkout"
      >
        kart
      </a>
    </button>
  </div>
</DocumentFragment>
`;

exports[`<ProductCard /> it should render the load version of the card 1`] = `
<DocumentFragment>
  <div
    class="css-1ooee9y-Card e1a1q5s70"
    data-testid="variants-card"
  >
    <div
      class="css-1px0wn7-Section eui42cl2"
    >
      <div
        class="css-1l33m80-SelectionSection eui42cl1"
      >
        <label>
          packing-size:
        </label>
        <select
          class="css-1nbhqyd-Select eui42cl0"
          name="ages"
        >
          <option
            data-testid="select-option"
            disabled=""
            selected=""
            value="sizes"
          >
            sizes
          </option>
        </select>
      </div>
    </div>
    <button
      class="css-hhtkyx-Button enhitpr0"
      name="button"
      type="button"
    >
      <a
        href="/checkout"
      >
        kart
      </a>
    </button>
  </div>
</DocumentFragment>
`;
